force-app/main/default/classes/NamingExclusions.cls

```diff
@@ -0,0 +1,134 @@
+        this.oldMap = oldMap != null ?
+                new Map<Id, Contact>(oldMap) : new Map<Id, Contact>();
+    }
+
+    public void synchronize() {
+        for (Contact con : contacts) {
+            INamingExclusionsString exclusionsAsString =
+                    new HouseholdNamingExclusionsString(con);
+            INamingExclusions exclusionsAsBooleans =
+                    new HouseholdNamingExclusionsCheckboxes(con);
+            if (!exclusionsAsString.equals(exclusionsAsBooleans)) {
+                mergeExclusions(exclusionsAsBooleans, exclusionsAsString);
+                setExclusionsStringFor(con, exclusionsAsString.value());
+                setExclusionsCheckboxes(con, exclusionsAsBooleans);
+            }
+        }
+    }
+
+    private void setExclusionsStringFor(Contact contact, String val) {
+        contact.npo02__Naming_Exclusions__c = val;
+    }
+
+    /**
+     * @param exclusionsAsBooleans Checkbox implemenation of Household Naming Exclusions
+     * @param exclusionsAsString Concatenated string implementation of Household Naming
+     * Exclusions
+     * @description Checks any exclusions checkboxes to true that are included in the
+     * concatenated string, and then updates the string to include exclusions (field names)
+     * for any Exclusion checkboxes that are marked TRUE for the Contact.
+     */
+    private void mergeExclusions(INamingExclusions exclusionsAsBooleans,
+            INamingExclusions exclusionsAsString) {
+        // Prioritize booleans by syncing from picklists first
+        exclusionsAsBooleans.synchronizeFrom(exclusionsAsString);
+        exclusionsAsString.synchronizeFrom(exclusionsAsBooleans);
+    }
+
+    private void setExclusionsCheckboxes(Contact con, INamingExclusions exclusions) {
+        con.Exclude_from_Household_Name__c = exclusions.isExcludedFromName();
+        con.Exclude_from_Household_Formal_Greeting__c = exclusions.isExcludedFromFormalGreeting();
+        con.Exclude_from_Household_Informal_Greeting__c = exclusions.isExcludedFromInformalGreeting();
+    }
+
+    public void syncNamingExclusionsBeforeInsert() {
+        for (Contact con : contacts) {
+            HouseholdNamingExclusionsString namingExclusionsString =
+                    new HouseholdNamingExclusionsString(con);
+
+            HouseholdNamingExclusionsCheckboxes exclusionsAsCheckboxes =
+                    new HouseholdNamingExclusionsCheckboxes(con);
+
+            if (namingExclusionsString.hasNamingExclusions()) {
+                exclusionsAsCheckboxes.updateFrom(namingExclusionsString);
+                setExclusionsCheckboxes(con, exclusionsAsCheckboxes);
+            }
+
+            if (exclusionsAsCheckboxes.hasNamingExclusions()) {
+                namingExclusionsString.synchronizeFrom(exclusionsAsCheckboxes);
+                setExclusionsStringFor(con, namingExclusionsString.value());
+            }
+        }
+    }
+
+    public void synchronizeBeforeUpdate() {
```

Comment:
To follow the same naming pattern, this method can be rename to `synchronizeExclusionsBeforeUpdate`